handsontable/src/core.js

```diff
@@ -2290,7 +2290,9 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
    * @fires Hooks#afterUpdateSettings
    */
   this.updateSettings = function(settings, init = false) {
-    const dataUpdateFunction = (firstRun ? instance.setData : instance.updateData).bind(this);
+    // TODO: uncomment the next line with the next major version update
+    // const dataUpdateFunction = (firstRun ? instance.setData : instance.updateData).bind(this);
+    const dataUpdateFunction = instance.setData.bind(this);
```

Comment:
```suggestion
    // TODO: uncomment the next line with the next major version update
    // Do not forget to re-enable the pending tests that cover the change:
    //  * https://github.com/handsontable/handsontable/blob/9f62c282a1c951b27cd8406aa27105bd32b05bb6/handsontable/test/e2e/core/toPhysicalColumn.spec.js#L70
    //  * https://github.com/handsontable/handsontable/blob/9f62c282a1c951b27cd8406aa27105bd32b05bb6/handsontable/test/e2e/core/toVisualColumn.spec.js#L70
    // const dataUpdateFunction = (firstRun ? instance.setData : instance.updateData).bind(this);
    const dataUpdateFunction = instance.setData.bind(this);
```
I bet that someone who would uncomment the breaking change line would forget that there are some tests for that. Especially that there are no failing tests after enabling the major change line. I suggest adding links to the pending tests, they are glued to the specific commit.

Reply:
Very true, good idea